### ISSUE TITLE
`comment_width` is ignored when bigger than `max_width`

### ISSUE THREAD

**Issue by konstin** [rust-lang/rustfmt#2939]
When `comment_width` is set to a larger value than `max_width`, `comment_width` is still bound by `max_width`. Take the following project:

**rustfmt.toml**

```toml
wrap_comments = true
comment_width = 1234
```

**main.rs**

```rust
/// Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore
fn main() {
    println!("Hello, world!");
}
```

The doc comment is fine according to the comment width, but still wrapped as it is longer than 100 characters:

```rust
/// Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt
/// ut labore
fn main() {
    println!("Hello, world!");
}
```

The use case I have for this is that I want to format only the code samples in the comments, but not comments itself. (I'm working on [pyo3](https://github.com/PyO3/pyo3) which has lots of wide doc comments that are wrapped incorrectly with the default `comment_width`, but which also have many code samples that should be properly formatted.)

`rustfmt -V`: `rustfmt 0.99.2-nightly (5c9a2b6 2018-08-07)`

**Comment by scampi** [comment]
I think if you remove `wrap_comments` and use the option `format_doc_comments`, you get what you want: it formats the doc test but leaves the comment alone.

**Comment by scampi** [comment]
fixed by #3535
